docs/how-to/push-update-on-mainnet.md

```diff
@@ -0,0 +1,120 @@
+Overview
+========
+
+Software update is a protocol mechanism that allows nodes to agree on software changes without altering protocol constants and update to this software version.
+1. Software update is proposed. Proposal is a datatype that gets into
+   the blockchain. It contains information about version changes and
+   hashes of update files.
+2. Update files are uploaded to the S3 bucket.
+3. Software update is confirmed by voting from majority of nodes.
+4. Nodes that see an update try to download and apply it
+   automatically.
+
+Now, again, step-by-step.
+
+Prerequisites
+=============
+1. Cluster nodes' keys. They are crucial to have since cluster nodes
+   are the only nodes that have stake in the system so they are the
+   only nodes that can propose an update and vote for it.
```

Comment:
Are all 7 core node keys required, or is a majority technically sufficient? Using all 7 for now is fine, but I'd like to understand the minimal technical requirement for the day when more nodes may be involved.

Reply:
Ah, this is answered on line 37. Only a majority of the secret keys are required.

Reply:
I suggest saying "Majority of cluster nodes' keys." here instead.

Reply:
The technical requirement is that the majority of _stake_ is required. I don't know how the stake is split between the seven secret keys, but I assume that the split is equal (I'll ask in Slack just in case).

Reply:
Ok, got it :)

My understanding from genesis creation is stake started out evenly distributed across these core nodes. I'm not sure whether that fluctuates as transactions are made.